### PR TITLE
[MODORDERS-803] - Provided error code for flow #3

### PR DESCRIPTION
**Pull request by Dmitriy-Butramyou**
## Purpose
Provide error code for flow 3

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/rest/core/exceptions/ErrorCodes.java
Comment (SerhiiNosko) [inline src/main/java/org/folio/rest/core/exceptions/ErrorCodes.java:108]: This name of error code is too generic, we need specific name to show user that related invoice lines contains lines which has status APPROVED for the current fiscal year.
PO_LINE_HAS_RELATED_APPRVOVED_INVOICE_ERROR("poLineHasRelatedApprovedInvoice");
Also please agree this name with @usavkov-epam

Comment (Dmitriy-Butramyou) [reply src/main/java/org/folio/rest/core/exceptions/ErrorCodes.java]: Name of error code updated

File: src/main/java/org/folio/service/invoice/PoLineInvoiceLineHolderBuilder.java
Comment (Abdulkhakimov) [inline src/main/java/org/folio/service/invoice/PoLineInvoiceLineHolderBuilder.java:58]: Please leave this method commented. I will uncomment it in scope of second flow. Thanks

**Review by Abdulkhakimov — APPROVED** [review]

File: src/main/java/org/folio/service/invoice/PoLineInvoiceLineHolderBuilder.java
Comment (SerhiiNosko) [reply src/main/java/org/folio/service/invoice/PoLineInvoiceLineHolderBuilder.java:58]: Can we merge PR for integration tests and unblock work for UI if leave this validation commented?

Comment (Abdulkhakimov) [reply src/main/java/org/folio/service/invoice/PoLineInvoiceLineHolderBuilder.java:58]: It's unreal from technically perspective

Comment (SerhiiNosko) [reply src/main/java/org/folio/service/invoice/PoLineInvoiceLineHolderBuilder.java:58]: okay, so you propose to merge this PR and related karate PR : https://github.com/folio-org/folio-integration-tests/pull/804  after Flow 2 PR will be merged

Comment (SerhiiNosko) [reply src/main/java/org/folio/service/invoice/PoLineInvoiceLineHolderBuilder.java:58]: okay so lets merge

**Pull request merged**
